modules/core/src/main/java/org/apache/ignite/internal/processors/platform/cluster/PlatformClusterNodeFilterImpl.java

```diff
@@ -48,21 +54,49 @@ public PlatformClusterNodeFilterImpl() {
      */
     public PlatformClusterNodeFilterImpl(Object pred, PlatformContext ctx) {
         super(pred, 0, ctx);
+
+        init();
     }
 
-    /** {@inheritDoc} */
-    @Override public boolean apply(ClusterNode clusterNode) {
+    /* Initializes the cluster node filter */
+    public void init(){
         try (PlatformMemory mem = ctx.memory().allocate()) {
             PlatformOutputStream out = mem.output();
 
             BinaryRawWriterEx writer = ctx.writer(out);
 
             writer.writeObject(pred);
+            out.synchronize();
+
+            handleId = ctx.gateway().clusterNodeFilterCreate(mem.pointer());
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean apply(ClusterNode clusterNode)  {
+
+        if(ctx == null){
+            return false;
```

Comment:
We should throw a descriptive exception instead. Using .NET node filter in mixed cluster is not supported. Silently ignoring this will lead to unexpected behavior.